src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -82,6 +84,39 @@ dual c = mempty
            invariant = fmap dualInvariant (invariant c)
          }
 
+-- Give the opposite initial state for the chosen signal
+bubbleInitialValue :: Eq a => a -> (a -> InitialValue) -> (a -> InitialValue)
+bubbleInitialValue s f y = if (y == s) then bubbleVal (f y) else f y
+  where
+    bubbleVal (Defined v) = Defined (not v)
+    bubbleVal x = x
+
+-- For the selected signal, toggle the directions of all transitions for this.
+bubbleCausality :: Eq a => a -> Causality (Transition a)
+                        -> Causality (Transition a)
+bubbleCausality s (Causality f t)
+    | signal t == s = Causality f (toggle t)
```

Comment:
Well, `a+ -> a-` should still be treated correctly even if it's not very meaningful :)